lib/ledger_sync/adaptors/quickbooks_online/bill_payment/ledger_serializer.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+require_relative '../currency/ledger_serializer'
+require_relative '../vendor/ledger_serializer'
+require_relative '../department/ledger_serializer'
+require_relative '../bill_payment_line_item/ledger_serializer'
+
+module LedgerSync
+  module Adaptors
+    module QuickBooksOnline
+      module BillPayment
+        class LedgerSerializer < QuickBooksOnline::LedgerSerializer
+          id
+
+          attribute ledger_attribute: 'TotalAmt',
+                    resource_attribute: :amount,
+                    type: LedgerSerializerType::AmountType
+
+          references_one ledger_attribute: :CurrencyRef,
+                         resource_attribute: :currency,
+                         serializer: Currency::LedgerSerializer
+
+          attribute ledger_attribute: 'VendorRef.value',
+                    resource_attribute: 'vendor.ledger_id'
+
+          attribute ledger_attribute: 'DepartmentRef.value',
+                    resource_attribute: 'department.ledger_id'
+
+          attribute ledger_attribute: 'APAccountRef.value',
+                    resource_attribute: 'account.ledger_id'
+
+          attribute ledger_attribute: 'DocNumber',
+                    resource_attribute: :reference_number
+
+          attribute ledger_attribute: 'PrivateNote',
+                    resource_attribute: :memo
+
+          attribute ledger_attribute: 'ExchangeRate',
+                    resource_attribute: :exchange_rate
+
+          attribute ledger_attribute: 'TxnDate',
+                    resource_attribute: :transaction_date,
+                    type: LedgerSerializerType::DateType
+
+          attribute ledger_attribute: 'PayType',
+                    resource_attribute: :payment_type,
+                    type: LedgerSerializerType::PaymentType
+
+          attribute ledger_attribute: 'CreditCardPayment' do |res|
+            if res.credit_card_account
```

Comment:
I wonder if we should use types for this, given we have to deserialize as well?  I am not sure block passing should exist and may just be a relic.  That said, I also don't know about the attribute structure here.  Once we dig into NS bill resource, we will have a clearer picture of the differences and how to model.

Reply:
True.

For `CreditCardPayment` inside there is just `CCAccountRef`. So its a simple reference.

For `CheckPayment` there is `BankAccountRef` and optional `PrintStatus` string attribute.

I may have to write type for it after all.